### PR TITLE
Feature/cleanup command

### LINKED ISSUE #37: Cleanup Command
Der Bot scheint nicht alle User aus der Datenbank zu entfernen.
Wir brauchen einen Command (Bot-Owner) der über die User iteriert und die nicht gefundenen User aus der Datenbank entfernt.

### PR DESCRIPTION
**Pull request by daspositioned**
closes #37

### CODE REVIEW

File: src/main/kotlin/com/github/devcordde/devcordbot/commands/owners/CleanupCommand.kt [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                logger.info { "Tag gelöscht: Author ${it.author}, Name: ${it.name}, Inhalt: ${it.content}" }
+                it.delete()
+            }
+        }
+    }
+
+    private fun isMemberOfGuild(guild: Guild, userID: Long): Boolean {
+        return guild.retrieveMemberById(userID).complete(true) != null
+    }
+}
```
Comment (jkoenig134) [inline src/main/kotlin/com/github/devcordde/devcordbot/commands/owners/CleanupCommand.kt]: newline

**Review by jkoenig134 — CHANGES_REQUESTED** [review]

**Review by jkoenig134 — APPROVED** [review]

**Review by jkoenig134 — APPROVED** [review]

**Review by DRSchlaubi — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/github/devcordde/devcordbot/commands/general/TagCommand.kt [hunk 2]
```diff
@@ -310,6 +315,27 @@ class TagCommand : AbstractCommand() {
...
+        override val displayName = "Cleanup"
+        override val description = "Entfernt die Level von ungültigen Accounts"
+        override val usage = "<tagname>"
+        override val permission = Permission.BOT_OWNER
+        override val commandPlace = CommandPlace.ALL
+
+        override suspend fun execute(context: Context) {
+            DevCordUser.all().forEach {
+                val member =
+                    context.jda.getGuildById(dotenv()["GUILD_ID"] ?: return@forEach)?.retrieveMemberById(it.userID)
```
[the remark this answers is not in the document]
Comment (DRSchlaubi) [reply src/main/kotlin/com/github/devcordde/devcordbot/commands/general/TagCommand.kt]: Also retrieving members 6k times will probably get us ratelimited

[the remark this answers is not in the document]
Comment (jkoenig134) [reply src/main/kotlin/com/github/devcordde/devcordbot/commands/general/TagCommand.kt]: In that point we shouldn´t use the cache. We want to see the really *current* status of the member.

[the remark this answers is not in the document]
Comment (DRSchlaubi) [reply src/main/kotlin/com/github/devcordde/devcordbot/commands/general/TagCommand.kt]: Using retrieve doesn help with that, cache is based on Gateway Events so the cache should be pretty up to date

Also we're just filtering if the member left so retrieving doesn't really help

Also also we,re retrieving member data twice per membe, dk about exact date limit in retrievemember but it's probably not intended to be called 4k Times within a few minutes

**Review by jkoenig134 — APPROVED** [review]

**Review by DRSchlaubi — APPROVED** [review]
LGTM

**Review by jkoenig134 — APPROVED** [review]

**Pull request merged**